### ISSUE TITLE
Cascade "data-snap-ignore" Elements

### ISSUE THREAD

**Issue by t0rpedo** [jakiestfu/Snap.js#32]
If using data-snap-ignore="true", then all child elements also require the attribute to be present for dragging to be disabled down the tree. This proves to be a bit tedious for many-child elements.

``` html
<div id="container" data-snap-ignore="true">
    <div id="content">This will still drag</div>
</div>

<div id="container" data-snap-ignore="true">
    <div id="content" data-snap-ignore="true">Now it won't drag</div>
</div>
```

To have the data-snap-ignore="true" cascade down from the parent element would be great!

Thus said, I love what you've done in such short time! Very cool stuff!

**Comment by jakiestfu** [comment]
How can I do this, do you know?

**Comment by racorbit** [comment]
Not the most elegant solution, but for the time being I just find all child elements and add the data attribute before initializing snapper.  A possibly more elegant solution would be to add an option for ignoring child elements of any element with the ignore attribute and then Apply the rule to the tree.  I need to investigate to find the best performing approach.
—
Rick

**Comment by lmartins** [comment]
This would very useful indeed. Having to replicate the  data-snap-ignore is really not convenient.

**Comment by t0rpedo** [comment]
After some pondering yesterday it seems like the best way to go about this would be, as racorbit noted, to indeed add a stop mechanism to the event listener assignments. This would cause that when a node is found with the no-drag parameter, the assignment iterator would skip to the next sibling without processing the marked node or it's children. 

The only negative aspect I could imagine at the moment would be the need for an additional function to re-add the events to the no-drag elements if you have interaction requirements, but that seems like an edge-case scenario. (However, now-a-days it seems like most edge-case situations become the norm fairly quickly.)

I would love to hear other options, but the above mentioned seems like it has the most potential.

EDIT: I just read the message thread for the referenced issue above. (#35) The idea of only targeting specific elements for the slide event has some potential!

**Comment by racorbit** [comment]
Yeah that's my thought exactly as I think it's somewhat of an edge case.  i am also nervous that it probably will be complained about though and thought of another option - to have two attributes, data-snap-ignore and data-snap-ignore-children or a more clever name that eludes me at the moment:)
—
Rick

**Comment by jakiestfu** [comment]
@racorbit `data-snap-ignore-children` is not a bad thought. Really, I want that to be implied with `data-snap-ignore`

I am going to look into adding live events so the listener works even on DOM reconstruction, and that would be the opportune time to look into better support for `data-snap-ignore`. Lets leave this issue open for a bit

**Comment by t0rpedo** [comment]
Here is a temporary quick fix for the current issue. Sorry it took me a while to post. My raw js skills aren't what they should be. JQuery has made me weak. (Yet so strong... :muscle:) Seeing as how snap.js was raw I intended for the solution to be so as well.

Just call the function `cascadeDragIgnore()` and children under any element with the `data-snap-ignore` attribute will also inherit it inline. 

:warning: Be sure to call it after any plugins that manipulate the DOM. Otherwise you may end up with children that don't have the attribute. Any dynamically generated elements stuff will also be neglected and will need to be dealt with upon creation. Like I said: Quick Fix

```
function cascadeDragIgnore() {
  // Default vars for snap.js
  var attribute = 'data-snap-ignore';
  var value = 'true';

  // Initial Vars
  var nodes = [];
  var allElements = document.getElementsByTagName('*');

  // Get all elements with 'data-snap-ignore' = true
  for (var i = 0; i < allElements.length; i++) {
    if (allElements[i].getAttribute(attribute) == value) {
      // Element exists with attribute. Add to array.
      nodes.push(allElements[i]);
    }
  }

  // For each parent
  for (var i = 0; i < nodes.length; i++) {
    if (nodes[i].nodeType === 1) {
        recurseAndChange(nodes[i]);
    }
  }  
}

function recurseAndChange(n){
    // start with the first
    n.setAttribute('data-snap-ignore', 'true');

    // Check for children
    if (n.hasChildNodes()) {
      // Start with first child       
      var child = n.firstChild;
      // Make sure we have a child       
      while(child){
        // Double check node type         
        if (child.nodeType === 1) {           
          recurseAndChange(child);         
        }
        // Move to sibling
        child = child.nextSibling;
      }
    }
}
```

**Comment by jakiestfu** [comment]
@t0rpedo I have a different solution.

Everytime you click an element, I just traverse upwards in the DOM in a loop until I find an element with my target attribute. If such an element exists, it will return it, but if not, it will return null, so this works, but I don't know how efficient it is. This will work for future elements, too, but again, imagine you have some deep seeded elements that have a parent a zillion nodes up, it may degrade performance. Thoughts? @racorbit 

Check out the following code here: http://jsfiddle.net/BWnx9/

``` javascript
function parentsUntil(el, attr) {
    while (el.parentNode) {
        if (el.getAttribute && el.getAttribute(attr)){
            return el;
        }
        el = el.parentNode;
    }
    return null;
}

var startDrag = function(e){

    var src = e.target ? e.target : e.srcElement,
        ignoreParent = parentsUntil(src, 'data-snap-ignore');

    if (ignoreParent) {
        return;
    }
    alert('Clicked');
};


document.addEventListener('click', startDrag);
```

**Comment by jakiestfu** [comment]
Fixed in 1.7.8, thanks for your help guys! 05dfa0bc629e22a61288bc7ba3760ef8d3984d3e
